packages/jit-html/src/template-compiler.ts

```diff
@@ -146,6 +169,110 @@ export class TemplateCompiler implements ITemplateCompiler {
     return compiledDefinition;
   }
 
+  private processLocalTemplates(
+    template: HTMLElement,
+    definition: CustomElementDefinition,
+    context: IContainer,
+    dom: IDOM
+  ) {
+    let root: HTMLElement | DocumentFragment;
+
+    if (template.nodeName === 'TEMPLATE') {
+      if (template.hasAttribute(localTemplateIdentifier)) {
+        throw new Error('The root cannot be a local template itself.'); /* TODO: use reporter */
+      }
+      root = (template as HTMLTemplateElement).content;
+    } else {
+      root = template;
+    }
+    const localTemplates = toArray(root.querySelectorAll('template[as-custom-element]')) as HTMLTemplateElement[];
+    const numLocalTemplates = localTemplates.length;
+    if (numLocalTemplates === 0) { return; }
+    if (numLocalTemplates === root.childElementCount) {
+      throw new Error('The custom element does not have any content other than local template(s).');
+    }
+    const localTemplateNames: Set<string> = new Set();
+
+    function getTemplateName(localTemplate: HTMLTemplateElement): string {
+      const name = localTemplate.getAttribute(localTemplateIdentifier);
+      if (name === null || name === '') {
+        throw new Error('The value of "as-custom-element" attribute cannot be empty for local template'); /* TODO: use reporter/logger */
+      }
+      if (localTemplateNames.has(name)) {
+        throw new Error(`Duplicate definition of the local template named ${name}`); /* TODO: use reporter/logger */
+      } else {
+        localTemplateNames.add(name);
+      }
+      return name;
+    }
+    function getBindingMode(bindable: Element): BindingMode {
+      switch (bindable.getAttribute(LocalTemplateBindableAttributes.mode)) {
+        case 'oneTime':
+          return BindingMode.oneTime;
+        case 'toView':
+          return BindingMode.toView;
+        case 'fromView':
+          return BindingMode.fromView;
+        case 'twoWay':
+          return BindingMode.twoWay;
+        case 'default':
+        default:
+          return BindingMode.default;
+      }
+    }
+
+    for (const localTemplate of localTemplates) {
+      if (localTemplate.parentNode !== root) {
+        throw new Error('Local templates needs to be defined directly under root.'); /* TODO: use reporter */
+      }
+      const name = getTemplateName(localTemplate);
+
+      // eslint-disable-next-line @typescript-eslint/class-name-casing
+      const localTemplateType = class { };
```

Comment:
```suggestion
      const localTemplateType = class LocalTemplate { };
      Reflect.defineProperty(localTemplateType, 'name', { value: pascalCase(name), configurable: true });
```

Just a bit of a nitpick, but would make it a bit more debugger friendly

Reply:
Done